packages/astro/src/types/public/integrations.ts

```diff
@@ -239,3 +239,8 @@ export interface AstroIntegration {
 		[K in keyof Astro.IntegrationHooks]?: Astro.IntegrationHooks[K];
 	} & Partial<Record<string, unknown>>;
 }
+
+/**
+ * A smaller version of the {@link RouteData} that is used in the integrations.
+ */
+export type IntegrationRouteData = Omit<RouteData, 'isIndex' | 'fallbackRoutes'>;
```

Comment:
Should `redirect` be of type `RouteData` for integrations?

Reply:
Good call. `redirectRoute` should be `IntegrationRouteData`, but `fallbackRoutes` has been excluded and not passed to the integrations